src/main/java/org/example/DatabaseHelper.java

```diff
@@ -0,0 +1,71 @@
+package org.example;
+
+import java.sql.*;
+import java.text.MessageFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/*
+In this database mySQL database with table name audit.
+The table contains following columns, path to the file, the word searched, the date time of the search
+with error and success messages
+ */
+public class DatabaseHelper {
+    /*
+      Storing data to database
+     */
+    public void storingDataToDataBase(String filepath, String WordSearch, String finalResult, int totalNumberOfWords, String errorMessage) throws SQLException {
+        Connection connectionToDataBase = null;
+        Statement statement;
+        String DateAndTime;
+        DateTimeFormatter dateAndTimeFormat = DateTimeFormatter.ofPattern(Constants.DATE_AND_TIME_FORMAT);
+        LocalDateTime now = LocalDateTime.now();
+        DateAndTime = dateAndTimeFormat.format(now);
+        try {
+            connectionToDataBase = connectionToDataBase();
+            statement = connectionToDataBase.createStatement();
+            DatabaseMetaData checkIfTableExists = connectionToDataBase.getMetaData();
+            ResultSet tables = checkIfTableExists.getTables(null, null, Constants.AUDIT, null);
+            if (tables.next()) {
+                String query = MessageFormat.format("INSERT INTO audit VALUES ({0},{1},{2},{3},{4},{5})", "'" + filepath + "'", "'" + WordSearch + "'", "'" + DateAndTime + "'", "'" + finalResult + "'", "'" + totalNumberOfWords + "'", "'" + errorMessage + "'");
```

Comment:
In the 'if' block, you are inserting data to the db.
In the 'else' block, you are calling saveDataToTable(), which also saves data to the db.

Rewrite this if-else condition.
Check if the table does not exist, if so, create the table. After that insert data to the table.

Reply:
Updated has suggested.